### ISSUE TITLE
Compile error with default attribute name

### ISSUE THREAD

**Issue by pity7736** [cython/cython#3735]
I have an error when use a default attribute name. I have this code:

```python
cdef class Field:

    _internal_type = None

    def __init__(self, default=None):
        self.default = default

    cpdef public parse(self, value):
        if value is None or type(value) is self._internal_type:
            return value
        return self._internal_type(value)


cdef class StrField(Field):

    _internal_type = str
```

**Comment by scoder** [comment]
Not sure yet, but you might have hit a bug, or at least an unfortunate language quirk.

`default` is a reserved name in C, and we normally escape those before writing them out into the C code. However, declaring the attribute as `public` prevents the internal escaping.
https://github.com/cython/cython/blob/2c7c22f51405299a4e247f78edf52957d30cf71d/Cython/Compiler/Symtab.py#L2229-L2230
What you probably intended was to get Python property accessors for the attribute, not make it publicly accessible for other C code. However, both unfortunately use the same language feature, the `public` modifier.

You can work around this problem by giving the attribute a different internal C name, e.g.:
`cdef public default "_default"`

**Comment by pity7736** [comment]
Yeah, my work around was to use `default_value`. It's not a big deal.
